overops-functions/src/main/java/com/takipi/udf/deployment/DeploymentRoutingFunction.java

```diff
@@ -73,95 +79,87 @@ private static String cleanPrefix(DeploymentRoutingInput input) {
 
 	private static void buildDeploymentRoutingViews(ContextArgs args, DeploymentRoutingInput input) {
 		ApiClient apiClient = args.apiClient();
-
 		String serviceId = args.serviceId;
+		int maxViews = input.max_views;
 
-		String categoryId = createDepCategory(apiClient, serviceId, input);
-
-		List<String> activeDeps = ClientUtil.getDeployments(apiClient, serviceId, true);
-		List<String> allDeps = ClientUtil.getDeployments(apiClient, serviceId, false);
-
-		if (activeDeps == null) {
-			System.err.println("Could not acquire active deps for service " + serviceId);
-			return;
-		}
+		Collection<SummarizedDeployment> deployments = ClientUtil.getSummarizedDeployments(apiClient, serviceId, false);
 
-		if (allDeps == null) {
-			System.err.println("Could not acquire all deps for service " + serviceId);
+		if (deployments == null) {
+			System.err.println("Could not acquire all deployments of service " + serviceId);
 			return;
 		}
 
 		Map<String, SummarizedView> views = ViewUtil.getServiceViewsByName(apiClient, serviceId);
 
 		if (views == null) {
-			System.err.println("Could not acquire view for service " + serviceId);
+			System.err.println("Could not acquire views of service " + serviceId);
 			return;
 		}
 
-		List<SummarizedView> activeDepViews = Lists.newArrayList();
-		List<SummarizedView> allDepViews = Lists.newArrayList();
+		// sort deployments by descending last_seen
+		List<SummarizedDeployment> sortedDeployments = deployments.parallelStream()
+				.sorted((dep1, dep2) ->
+						Long.compare(
+								ISODateTimeFormat.dateTimeParser().parseDateTime(dep2.last_seen).getMillis(),
+								ISODateTimeFormat.dateTimeParser().parseDateTime(dep1.last_seen).getMillis()))
+				.collect(Collectors.toList());
 
-		for (SummarizedView view : views.values()) {
-			String viewDepName;
-
-			if (input.prefix != null) {
-				viewDepName = view.name.replace(cleanPrefix(input), "");
-			} else {
-				viewDepName = view.name;
-			}
+		// map deployments to viewIds of their corresponding views
+		Map<String, String> deploymentViewIdMap = Maps.newHashMapWithExpectedSize(maxViews);
 
-			if (activeDeps.contains(viewDepName)) {
-				activeDepViews.add(view);
-			} else if (allDeps.contains(viewDepName)) {
-				allDepViews.add(view);
+		for (SummarizedView view : views.values()) {
+			String deploymentNameFromView = (input.prefix != null) ?
```

Comment:
maybe an easier approach is to search for a deployment suffix in a "view name" and not trim the view's name?

Reply:
That way will match views with other prefixes.

BTW, `viewName.startsWith(prefix) && viewName.endsWith(depName)`
is also not accurate because there can be anything in between.